core/src/deck.test.ts

```diff
@@ -0,0 +1,29 @@
+import test from "node:test";
+import assert from "assert";
+import { MonetaryValue } from "./monetaryvalue";
+import { Card } from "./card";
+import { Deck } from "./deck";
+
+test("Deck", async (t) => {
+  const card1 = new Card(1, 0, new MonetaryValue("red", 1), new MonetaryValue().add("blue", 1).add("white", 1).add("red", 4));
+  const card2 = new Card(1, 1, new MonetaryValue("yellow", 1), new MonetaryValue().add("blue", 1).add("white", 1).add("yelloe", 4));
+  const card3 = new Card(1, 0, new MonetaryValue("blud", 1), new MonetaryValue().add("blue", 3).add("white", 1).add("red", 2));
+  const deck = new Deck([card1, card2, card3]);
```

Comment:
If you wrap these instantiations inside a `beforeEach()` then the test cases don't need to worry about modifying data that other test cases are using.

For example, in the future if you add another test that calls `deck.draw()` (or something else that modifies the size of the deck), it would be difficult to determine what exactly the size of the deck _should_  be.

Reply:
I use context to store the reference in beforeEach, any more convenient way to do that XD?

Reply:
I guess you could do something like this:

```
test("Deck", async (t) => {
  let card1: Card, card2: Card, card3: Card;
  let deck: Deck;
  
  beforeEach(() => {
    card1 = new Card(...)
    ....
    deck = new Deck(...)
  })

  await t.test("draw", async() => {
    ...
  })
})
```

Btw I think `beforeEach()` should go inside of `test()`?

Reply:
Am I fooled by Copilot...?
<img width="757" alt="圖片" src="https://github.com/qoosuperman/panoply/assets/50241297/58fc1280-3189-47bc-86a6-82721f4b1031">

Reply:
[updated](https://github.com/qoosuperman/panoply/pull/8/commits/c7521631dc7df8a12c45bf63c7561201b7de25a4#diff-3278903c5b00723ed7d745dbc722be76be8689b8de421a8de8e55851f4de57c7R18-R28)

Reply:
What it's referring to as the "test case" is probably `t.test("draw", async () => { ... })`, not the outer `test()`?

Reply:
https://nodejs.org/api/test.html#beforeeachfn-options

Reply:
So I just followed the instruction provided by it XD. Anyway, I prefer what you say.

Reply:
Lol aite